File: solutions/csharp/pythagorean-triplet/PythagoreanTriplet.cs
using System.Collections;

public static class PythagoreanTriplet
{
    public static IEnumerable<(int a, int b, int c)> TripletsWithSum(int sum)
    {
        var triplets = new List<(int a, int b, int c)>();
        for (int i = 0; i <= sum; i++)
        {
            for (int j = 0; j <= sum - i; j++)
            {
                int k = sum - i - j;
                if (i < j && j < k)
                {
                    if (i * i + j * j == k * k)
                    {
                        triplets.Add((i, j, k));
                    }
                }
            }
        }
        return triplets;
    }
}

File: solutions/python/pythagorean-triplet/pythagorean_triplet.py
def triplets_with_sum(number):
    triplets = []

    for c in range(3, number + 1):
        for b in range((number - c) / 2, number - c + 1):
            a = number - b - c
            if a < b and b < c and a**2 + b**2 == c**2:
                triplets += [[a, b, c]]

    return triplets
